sdk/highlight-node/src/client.ts

```diff
@@ -212,4 +258,36 @@ export class Highlight {
 			.forceFlush()
 			.catch((e) => console.warn('highlight-node failed to flush: ', e))
 	}
+
+	async waitForFlush() {
+		return new Promise<void>(async (resolve) => {
+			let resolved = false
+			let waitingForFinishedSpans = false
```

Comment:
This logic waits for finished spans to appear, and finishes itself on the next interval.

This logic may not be strictly necessary. It MAY be possible to finish as soon as finished spans have been identified.